qemu/qmp.go

```diff
@@ -542,13 +543,13 @@ func QMPStart(ctx context.Context, socket string, cfg QMPConfig, disconnectedCh
 		return nil, nil, fmt.Errorf("Canceled by caller")
 	case <-disconnectedCh:
 		return nil, nil, fmt.Errorf("Lost connection to VM")
-	case version = <-connectedCh:
+	case q.version = <-connectedCh:
 		if version == nil {
```

Comment:
This should be if q.version == nil.

Also you need to delete the var version *QMPVersion above which is no longer used.